confmap/converter/expandconverter/expand.go

```diff
@@ -90,19 +91,6 @@ func (c converter) expandEnv(s string) (string, error) {
 			return "$"
```

Comment:
Also, we need to mention that using `expanconverter` will cause double escaping `$$$$`->`$` or we can change this line to return `$$`. This comment should probably be addressed in https://github.com/open-telemetry/opentelemetry-collector/pull/10508

Reply:
Good point. I'll add it here because it is a result of using the now deprecated expandconverter.

Reply:
Updated

Reply:
Thinking more about it, maybe we should change it to `return "$$"`. I don't see any problems with that, and we keep the converter functional as before. WDYT?

Reply:
That assumes you're using the converter as part of our `confmap.Resolver`.  I'll update the comment to clarify the double escaping only happens when used with `confmap.Resolver`.

Reply:
Is there a way to use it without the resolver?

Reply:
It is an independent package so technically anyone could pull it in and do something with it. That is, granted, an unlikely scenario.

Reply:
Ok, let's keep it as is 👍